### ISSUE TITLE
Provider v3.8.0 misbehaving on resource/aws_organizations_policy -> newly introduced tag feature

### ISSUE THREAD

**Issue by krathow** [terraform-providers/terraform-provider-aws#15347]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version
Terraform v0.13.3
+ provider registry.terraform.io/hashicorp/aws v3.8.0

### Affected Resource(s)

* aws_organizations_policy

### Terraform Configuration Files

```hcl
resource "aws_organizations_policy" "security_scp" {
  content = jsonencode(
    {
      "Version" : "2012-10-17",
      "Statement" : [
        {
          "Sid" : "DenyRegions",
          "Effect" : "Deny",
          "Action" : "*",
          "Resource" : "*",
          "Condition" : {
            "StringNotEquals" : {
              "aws:RequestedRegion" : [
                "eu-central-1",
                "us-east-1",
                "eu-west-1"
              ]
            },
            "ArnNotLike" : {
              "aws:PrincipalArn" : [
                "arn:aws:iam::*:role/SharedServiceAccessRole"
              ]
            }
          }
        },
        {
          "Sid" : "Stmt123333333",
          "Effect" : "Deny",
          "Action" : [
            "aws-portal:Modify*",
            "aws-portal:ModifyAccount",
            "aws-portal:ModifyBilling",
            "aws-portal:ModifyPaymentMethods",
            "cloudtrail:DeleteTrail",
            "cloudtrail:RemoveTags",
            "cloudtrail:StartLogging",
            "cloudtrail:StopLogging",
            "cloudtrail:DescribeTrails",
            "cloudtrail:UpdateTrail",
            "ec2:DeleteFlowLogs",
            "iam:AddClientIDToOpenIDConnectProvider",
            "iam:CreateLoginProfile",
            "iam:CreateOpenIDConnectProvider",
            "iam:CreateSAMLProvider",
            "iam:DeleteAccountPasswordPolicy",
            "iam:DeleteSAMLProvider",
          ],
          "Resource" : [
            "*"
          ],
          "Condition" : {
            "ArnNotLike" : {
              "aws:PrincipalArn" : [
                "arn:aws:iam::*:role/SharedServiceAccessRole"
              ]
            }
          }
        },
        {
          "Sid" : "DenyConfig",
          "Effect" : "Deny",
          "Action" : [
            "config:DeleteConfigRule",
            "config:DeleteConfigurationRecorder",
            "config:DeleteDeliveryChannel",
            "config:StopConfigurationRecorder"
          ],
          "Resource" : [
            "arn:aws:cloudformation:*:*:stack/StackSet-ZF-*"
          ],
          "Condition" : {
            "ArnNotLike" : {
              "aws:PrincipalArn" : [
                "arn:aws:iam::*:role/SharedServiceAccessRole"
              ]
            }
          }
        }
      ]
    }
  )
  description = "Block Services for accounts"
  name        = "Security-SCP"
  type        = "SERVICE_CONTROL_POLICY"
}
```

### Debug Output

Error: error listing tags: InvalidInputException: You provided a value that does not match the required pattern.
{
  RespMetadata: {
    StatusCode: 400,
    RequestID: "07b02603-de9b-48a7-a6c2-c5870177cd3e"
  },
  Message_: "You provided a value that does not match the required pattern.",
  Reason: "INVALID_PATTERN"
}

**Comment by gdavison** [comment]
Thanks for raising this issue, @krathow. Are you seeing this error creating a new `aws_organizations_policy`, updating an existing resource, or during a `terraform import`?

To help us analyze what's happening, can you include the debug output of your `terraform apply`? See https://www.terraform.io/docs/internals/debugging.html to enable debug output. Please create a Github Gist to hold the debug output and put a link to the Gist here.

**Comment by krathow** [comment]
Hey @gdavison ,

I am seeing the error on a `terraform plan` on a workspace in Terraform Enterprise.
We imported the organizations policies into terraform successfully with AWS-provider 3.7.
And if we now just upgrade the AWS-provider to 3.8 and run a plan it fails.

See here the redacted gist (I can also upload the full output against the hashicorp-case if that helps):

https://gist.githubusercontent.com/krathow/9e0a2165cf4d29bb59324e7e19e9a86b/raw/16c1f4552eb8c147745fc0cd421aa9bc8b903c4d/orgnaization_policies_error.debug

Thanks for your help!

**Comment by mpailloncy** [comment]
FWIW, getting the exact same error with 3.8.0. Working properly with 3.7.0 as well

**Comment by gdavison** [comment]
Unfortunately, that debug output doesn't include the requests that are returning the error. Terraform is querying a few different resources concurrently, so the output is interleaved. The request and response are for the API call `ListTagsForResource`.

If you've opened a support case with us, I can work with the support team to track it issue down, otherwise we can continue investigating here

**Comment by krathow** [comment]
Hey @gdavison - 
I've uploaded the full debug output against the case.
Please let me know if I can provide more to help.

Would be great to have a script/settings to remove confidential output for working better with the community.

**Comment by gdavison** [comment]
Some sort of redaction mode would be very useful, especially with all of the ARNs written to logs. We have some proposals circulating now for managing sensitive values, but I don't know of any planned release schedule.

I've looked through the log, and it looks like the provider is failing when it tries to read tags from the Organization policy with ID `p-FullAWSAccess`. This is an AWS-managed policy, so tags are not supported on them. AWS-managed resources shouldn't be imported into Terraform, since they cannot be changed.

To resolve this you should update your configuration to directly reference the policy ID, either as a string or by defining a `local` value. Something like

```terraform
locals {
  organization_policy_full_access_id = "p-FullAWSAccess"
}

resource "aws_organizations_policy_attachment" "account" {
  policy_id = local. organization_policy_full_access_id
  target_id = "123456789012"
}
```

You may also need to remove the `aws_organizations_policy` resource from your Terraform state, using [`terraform state rm`](https://www.terraform.io/docs/commands/state/rm.html).

If I've assumed incorrectly, and the AWS-managed policy wasn't imported, I will probably need to see more of your Terraform configuration. The configuration can be added to the support case as well.

**Comment by krathow** [comment]
Hey, 

with that trick it works!

Would be great to have some warning in the documentation.

Thanks for your help and time!

**Comment by gdavison** [comment]
I've created #15397 to not throw an error if it has been imported and prevent importing in the future. It didn't make it in to 3.9.0, but we expect it to be released in 3.10.0

**Comment by krathow** [comment]
thank you very much @gdavison
